.github/workflows/run-test-coverage.yml

```diff
@@ -0,0 +1,41 @@
+# This workflow will install Python dependencies, run tests, generate coverage report and upload it to codecov
+# For more information see: https://help.github.com/actions/language-and-framework-guides/using-python-with-github-actions
+
+name: build
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+jobs:
+  coverage:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    - name: Set up Python 3.x
+      uses: actions/setup-python@v2
+      with:
+        python-version: '3.8'
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        pip install -r requirements-dev.txt
+        pip install pytest-cov
+    - name: Install ruptures
+      run: |
+        python setup.py install
+    - name: Test with pytest
+      run: |
+        pytest --cov-report=xml --cov=ruptures tests/
+
+    - name: Upload to Codecov
+      env:
+        CODECOV_TOKEN: ${{ secrets.CODECOV_TOKEN }}
```

Comment:
This is not required for public repo (see [here](https://github.com/codecov/codecov-action)).